lunr/query_lexer.py

```diff
@@ -1,6 +1,15 @@
+from typing import Callable, List, TypedDict
 from lunr.tokenizer import default_separator
 
 
+# FIXME: Could use a dataclass or a NamedTuple
+class Lexeme(TypedDict):
```

Comment:
Everything in lunr.py is very dict heavy as it was easier to port from Javascript that way. A dataclass sounds like a good fit but probably best to tackle in a different PR.

Reply:
Understood!  There isn't any performance advantage to dataclasses though they give a more Javascript-like API.  TypedDict at least allows for type checking!  I'll remove the comment.